src/nexus/backend/model_ins.h

```diff
@@ -116,6 +116,8 @@ class ModelInstance {
    * \param task Pointer to task.
    */
   virtual void Postprocess(std::shared_ptr<Task> task) = 0;
+  /*! \brief Get peak memory usage. */
+  virtual int64_t get_peak_bytes_in_use() { return -1; }
```

Comment:
UpperCamelCase as well.

I don't think returning magic number is a good idea. But since we only support TensorFlow now, I guess we also don't need to worry about not implemented case either. I'd recommend just use `LOG(FATAL)` to terminate the whole process if the derived class doesn't support it.

Also, please move the implementation to `.cpp` and leave only the declaration in `.h`.

Reply:
Done.